app/subscriber/src/features/my-reports/admin/ReportAdmin.tsx

```diff
@@ -63,38 +65,50 @@ export const ReportAdmin: React.FC<IReportAdminProps> = ({ path: defaultPath = '
 
   const handleSubmit = React.useCallback(
     async (values: IReportForm) => {
-      const originalId = values.id;
-      const report = originalId ? await updateReport(values) : await addReport(values);
-
-      if (!originalId) {
-        navigate(`/reports/${report.id}${path ? `/${path}` : ''}`);
-        toast.success(`Successfully created '${report.name}'.`);
-      } else {
-        setReport({
-          ...report,
-          sections: report.sections.map((section, index) => ({
-            ...section,
-            expand: values.sections[index].expand,
-          })),
-          hideEmptySections: getHideEmpty(report.sections),
+      try {
+        const sameNameReport = await findReports({
```

Comment:
When someone attempts to save a report with the same name it should return a message you can parse to determine if a report already exists with the name.  

Doing it this way requires an additional request to the API before submitting.

Can you see how it would work with the original error response and parsing the error?

Reply:
I haven't came into a final conclusion yet, but with a few minutes talking with Jonny, I couldn't intercept the global toast error, and the error I can capture at the component level is more generic, without the Duplicate Key Info. (That was my very initial goal ... but I'm really struggling with it)